part_2/second_split/method_v2.R

```diff
@@ -0,0 +1,398 @@
+rm(list = ls())
+setwd("/Volumes/GoogleDrive/My Drive/alsta_analysis_LA/second_split")
+
+######################################
+##### Section 1: Import packages #####
+######################################
+library(forecast)
+library(bsts)
+library(glmnet)
+library(plyr)
+library(lme4)
+library(strucchange)
+library(changepoint)
+library(mice)
+source("functions_v2.R")
+
+####################################
+##### Section 2: Preprocessing #####
+####################################
+
+### Impute data ###
+# Remove future data.
+end_date <- as.POSIXlt("2018-06-21", tz = "America/Los_Angeles")
+
+# Impute range: 2017-08-01 to 2017-09-01.
+# Read the original data to get the memory information.
+data_agg_sum <- readRDS("data_agg_sum.rds")
+data_impute_prep_mem <- data_agg_sum[data_agg_sum$date < end_date,]
+data_impute_prep_mem[data_impute_prep_mem$memory_gib_seconds > 5e4,] # 2018-05-01
+data_agg_sum[as.character(data_agg_sum$date) == "2018-05-01", "memory_gib_seconds"] <- mean(data_agg_sum[as.character(data_agg_sum$date) == "2018-04-30", "memory_gib_seconds"],
+                                                                                            data_agg_sum[as.character(data_agg_sum$date) == "2018-05-02", "memory_gib_seconds"])
```

Comment:
These lines are too wide.

Reply:
I made it shorter. If you still feel it is not fine, I can future break out the line by storing the values before the day and after the day separately.